ksql-engine/src/main/java/io/confluent/ksql/internal/KsqlEngineMetrics.java

```diff
@@ -41,8 +48,86 @@ public KsqlEngineMetrics(String metricGroupPrefix, KsqlEngine ksqlEngine) {
     this.ksqlEngine = ksqlEngine;
 
     this.metricGroupName = metricGroupPrefix + "-query-stats";
-    this.numActiveQueries = metrics.sensor(metricGroupName + "-active-queries");
-    numActiveQueries.add(
+
```

Comment:
Unit tests for this class?

Reply:
This class is basically glue code: it creates a bunch of sensors and then just calls 'record' on them. All the logic is in `MetricCollectors`. The corresponding tests are in `MetricCollectorsTest`. 

However, since it does manage sensor lifecycle, I added a test to ensure that all the sensors owned by the class are removed on close. 

Let me know if there are any other productive tests which can be written for this.

Reply:
Actually I thought about it a bit more and think that adding tests for the plumbing can be useful.. one can change metric names, etc. and having tests to catch it useful, because those changes will eventually break the dashboards built on the metrics.

I added a bunch of test cases to make sure that the metrics are being reported with the expected names. If the metric name conventions change in the future, we will catch them.